contributed_definitions/nyaml/NXopt.yaml

```diff
@@ -56,348 +43,675 @@ NXopt(NXobject):
           vs. the normal of the bottom reflective (substrate) surface in the
           sample if not equal to the angle(s) of incidence.
+          These two directions span the plane of detection.
         dimensions:
           rank: 1
           dim: [[1, N_detection_angles]]
-      sample_stage(NXsubentry):
+      angle_of_incident_and_detection_beam(NX_NUMBER):
+        exists: optional
+        unit: NX_ANGLE
         doc: |
-          Sample stage, holding the sample at a specific position in X,Y,Z
-          (Cartesian) coordinate system and at an orientation defined
-          by three Euler angles (alpha, beta, gamma).
-        stage_type:
-          doc: |
-            Specify the type of the sample stage.
-          enumeration: [manual stage, scanning stage, liquid stage, gas cell, cryostat]
-        alternative:
+          Angle between the incident and detection beam.
+          If angle_of_detection + angle_of_incidence = angle_of_incident_and_detection_beam,
+          then the setup is a reflection setup.
+          If angle_of_detection + angle_of_incidence != angle_of_incident_and_detection_beam
+          then the setup may be a light scattering setup.
+          (i.e. 90° + 90° != 90°, i.e. incident and detection beam in the sample surface, but
+          the angle source-sample-detector is 90°)
+
+        dimensions:
+          rank: 1
+          dim: [[1, XX]]
+        \@units:
+      angle_of_in_plane_sample_rotation(NX_NUMBER):
+        exists: optional
+        unit: NX_ANGLE
+        doc: |
+          Angle of the inplane orientation of the sample. This might be an arbitrary,
+          angle without specific relation to the sample symmetry,
+          of the angle to a specific sample property (i.e. crystallographic axis or sample
+          shape such as wafer flat)
+        dimensions:
+          rank: 1
+          dim: [[1, XX]]
+      lateral_focal_point_offset(NX_NUMBER):
+        exists: optional
+        unit: NX_LENGTH
+        doc: |
+          Specify if there is a lateral offset on the sample surface, between the focal 
+          points of the incident beam and the detection beam.
+      # beam path elements
+      source_TYPE(NXsource):
+        exists: [min, 1, max, infty]
+        type:
+          exists: recommended
+          enumeration: [laser, dye-laser, broadband tunable light source, X-ray Source, other]
+        type_other:
           exists: optional
           doc: |
-            If there is no motorized stage, we should at least qualify where
-            the beam hits the sample and in what direction the sample stands
-            in a free-text description, e.g. 'center of sample, long edge
-            parallel to the plane of incidence'.
-        environment_conditions(NXenvironment):
+            Specification of type, may also go to name.
+        name:
+          exists: recommended
+        source_polarization:
```

Comment:
This is a beam property, right? You don't need this here.

Reply:
Yea, its redundent.